src/lambda/handler-runner/docker-runner/DockerRunner.js

```diff
@@ -15,7 +15,7 @@ export default class DockerRunner {
       provider,
     } = funOptions
 
-    this.#codeDir = codeDir
+    this.#codeDir = dockerOptions.hostServicePath || codeDir
```

Comment:
If the function has [`artifact` option](https://www.serverless.com/framework/docs/providers/aws/guide/packaging#artifact), serverless-offline will extract artifact to `codeDir` (in tmpdir) and mount it into container.

https://github.com/dherault/serverless-offline/blob/74217253628083d3031750fd7d5b53361173522f/src/lambda/LambdaFunction.js#L91-L105

https://github.com/dherault/serverless-offline/blob/74217253628083d3031750fd7d5b53361173522f/src/lambda/LambdaFunction.js#L219-L242

I think overwriting `codeDir` by `dockerOptions.hostServicePath` all the time will break this behavior.

Could you ensure that `dockerOptions.hostServicePath` overwrite only if `codeDir` matches [`servicePath` in `funOptions`](https://github.com/dherault/serverless-offline/blob/74217253628083d3031750fd7d5b53361173522f/src/lambda/LambdaFunction.js#L116)?

```suggestion
    this.#codeDir = codeDir
    if (dockerOptions.hostServicePath && this.#codeDir === funOptions.servicePath) {
        this.#codeDir = dockerOptions.hostServicePath
    }
```